pulp_rpm/app/viewsets/package.py

```diff
@@ -93,15 +93,19 @@ def create(self, request):
             return super().create(request)
 
         # signing case
-        request.data.pop("file")
         validated_data = serializer.validated_data
-        temp_uploaded_file = validated_data["file"]
         signing_service_pk = validated_data["repository"].package_signing_service.pk
         signing_fingerprint = validated_data["repository"].package_signing_fingerprint
+        if "file" in validated_data:
+            request.data.pop("file")
+            temp_uploaded_file = validated_data["file"]
+            pulp_temp_file = PulpTemporaryFile(file=temp_uploaded_file.temporary_file_path())
+            pulp_temp_file.save()
+        else:
+            pulp_temp_file = validated_data["upload"]
+            uploaded_package = Upload.objects.get(pk=pulp_temp_file.pk)
```

Comment:
This line looks unused. Is there a reason to keep it here?

```python
uploaded_package = Upload.objects.get(pk=pulp_temp_file.pk)
```

Reply:
Nope, this was used in testing and is completely unnecessary.  Thanks for catching it!